### PR TITLE
Add featureBranchRegex to feature-start mojo

### PR DESCRIPTION
**Pull request by steppinrazor**
- By default regex is `.*` and allows all input

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -230,5 +230,11 @@
             <version>2.5.3</version>
             <scope>compile</scope>
         </dependency>
+        <dependency>
+            <groupId>org.testng</groupId>
```
Comment (aleksandr-m) [inline pom.xml]: Why testng? Can you use junit? Or I can change it to junit after the merge.

File: src/main/java/com/amashchenko/maven/plugin/gitflow/GitFlowFeatureStartMojo.java [hunk 2]
```diff
@@ -66,6 +76,11 @@ public void execute() throws MojoExecutionException, MojoFailureException {
                             .prompt("What is a name of feature branch? "
                                     + gitFlowConfig.getFeatureBranchPrefix());
 
+                    if(!isBranchNameRegexCompliant(featureName)){
```
Comment (aleksandr-m) [inline src/main/java/com/amashchenko/maven/plugin/gitflow/GitFlowFeatureStartMojo.java]: Maybe change default to blank and call `matches` only if it is not blank?

File: src/test/java/com/amashchenko/maven/plugin/gitflow/GitFlowFeatureStartMojoTest.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.*;
+
+/**
+ * Created by krs on 2/18/17.
+ */
+public class GitFlowFeatureStartMojoTest {
```
Comment (aleksandr-m) [inline src/test/java/com/amashchenko/maven/plugin/gitflow/GitFlowFeatureStartMojoTest.java]: On closer look. This test as it is now is really don't needed, essentially it tests only `Pattern.matches` and not the logic in `GitFlowFeatureStartMojo`.

Comment (steppinrazor) [inline src/test/java/com/amashchenko/maven/plugin/gitflow/GitFlowFeatureStartMojoTest.java]: The test is in `GitFlowFeatureStartMojo` and cannot be tested without an instance of that class in the same package scope so it only seemed natural - more tests could be added in the future.

File: pom.xml (hunk 1 above)
Comment (steppinrazor) [inline pom.xml]: I use TestNg to use its `DataProvider` interface.

File: src/main/java/com/amashchenko/maven/plugin/gitflow/GitFlowFeatureStartMojo.java (hunk 2 above)
Comment (steppinrazor) [inline src/main/java/com/amashchenko/maven/plugin/gitflow/GitFlowFeatureStartMojo.java]: I wrapped everything in a `if(isNotBlank(...)) {...}`

**Comment by aleksandr-m** [comment]
Implemented in d771d91e28ac3b031a0e969d78b3fdf58cd3b13f.

**Pull request closed without merging**